### PR TITLE
[WiP] Fixing Image Downloads

### PR DESCRIPTION
**Pull request by vanitasvitae**
* [x] Fixed faulty extensions on downloaded images (#102)
* [ ] Fixed gifs being shared as static images

This is done in a very hacky way. We should consider to rework image downloading/sharing etc. completely. (Also with ImageDownloadNotifications and ImageViewers in mind (#107))

### CODE REVIEW

**Comment by vanitasvitae** [comment]
I'd propose to ditch the DownloadManager completely and to download all images via InputStreams. Images needed in the app (eg. profile picture) can stay how they are atm. Images the user wants to download will be saved to disk (and in case of sharing they get shared from there). We can implement the determination of the filetype by comparing leading bytes (like I did), but using a PushbackInputStream, to take a look at the first bytes.
After we determined the filetype, we can then go ahead and download the image (either to a bitmap, or to a gif), save it to memory and show a notification or share it to another app.

**Comment by gsantner** [comment]
Isssue no102 merged into #107

File: app/src/main/java/com/github/dfa/diaspora_android/web/ContextMenuWebView.java [hunk 1]
```diff
@@ -107,17 +116,9 @@ public void onClick(DialogInterface dialog, int which) {
...
-                                File destinationFile = new File(Environment.getExternalStorageDirectory() + "/Pictures/Diaspora/"
-                                        + System.currentTimeMillis() + ".png");
-                                request.setDestinationUri(Uri.fromFile(destinationFile));
-                                ((DownloadManager) context.getSystemService(Context.DOWNLOAD_SERVICE)).enqueue(request);
-
-                                Toast.makeText(context, context.getText(R.string.share__toast_saved_image_to_location) + " " +
-                                        destinationFile.getAbsolutePath(), Toast.LENGTH_LONG).show();
-                            }
+                            SaveImageTask saveTask = new SaveImageTask();
+                            saveTask.setContext(context);
```
Comment (gsantner) [inline app/src/main/java/com/github/dfa/diaspora_android/web/ContextMenuWebView.java:120]: Constructor (Context context)

File: app/src/main/java/com/github/dfa/diaspora_android/service/SaveImageTask.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                        //GIF
+                        if (new String(a).startsWith("GIF")) {
+                            AppLog.d(this, "is gif");
+                            extension = ".gif";
+                        } else {
+                            AppLog.d(this, "is SPARTAAAA! (GIF)");
+                        }
+                } catch (IOException ignored) {
+                }
+                File destinationFile = new File(Environment.getExternalStorageDirectory() + "/Pictures/Diaspora/"
```
Comment (gsantner) [inline app/src/main/java/com/github/dfa/diaspora_android/service/SaveImageTask.java:61]: = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES), "dandelion");

**Comment by gsantner** [comment]
Has this PR something usable, or complete rework anyway? Can be closed?

**Comment by vanitasvitae** [comment]
@gsantner I'll have to revisit this once I have time again.

**Pull request closed without merging**